src/CMakeLists.txt

```diff
@@ -84,12 +81,20 @@ if(CMAKE_INTERPROCEDURAL_OPTIMIZATION)
     # Policy that enables INTERPROCEDURAL_OPTIMIZATION
     set(CMAKE_POLICY_DEFAULT_CMP0069 NEW)
     cmake_policy(SET CMP0069 NEW)
-    set(CMAKE_INTERPROCEDURAL_OPTIMIZATION ON)
     if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
       # Clang has some issues with thinLTO (segfaults on final linking)
       # This enables full LTO through CMake
       # FullLTO is slow, but may yield performance improvements.
-      add_compile_options(-flto=full)
+      #set(CMAKE_{lang}_COMPILE_OPTIONS_IPO ${CMAKE_CXX_COMPILE_OPTIONS_IPO} -flto)
+      add_compile_options(-flto) # alternate way of setting LTO.
+    elseif(CMAKE_CXX_COMPILER_ID MATCHES "GNU")
+      # If we are on GCC, override CMake do parallel LTO.
+      include(ProcessorCount)
+      ProcessorCount(N_LTO_JOBS)
```

Comment:
As I mentioned previously, the problem with this is that you'll suddenly need 30 or more GB RAM for the build -- on 8 cores it can happen that a 8-parallel LTO job will run together with 7 pybind compilations and each of those alone can easily take two gigabytes of memory. On the CI it could result in flaky builds, where they randomly fail due to running out of memory.

Two options:

- Make the job count configurable by passing a number to the option (so people could do `--lto=$(nproc)` or `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=$(nproc)`, if they have enough RAM) and then you extract that value here and pass it to `-flto=`; `--lto` alone or `-DCMAKE_INTERPROCEDURAL_OPTIMIZATION=ON` would mean serial LTO.
- Drop this special case altogether for now.

Another way to fix this would be using [Ninja job pools](https://cmake.org/cmake/help/latest/prop_gbl/JOB_POOLS.html), where the memory-hungry jobs wouldn't run in parallel, but I didn't investigate this thoroughly yet and fear that it would slow down the regular non-LTO build too much.

Reply:
Dropping this is fine by my. I added it per your earlier suggestion.